internal/controllers/analyzer/analyzer.go

```diff
@@ -457,3 +469,60 @@ func (a *Analyzer) isWarning(err string) bool {
 		strings.Contains(err, messages.MsgWarnBrakemanNotRubyOnRailsProject) ||
 		strings.Contains(err, messages.MsgWarnGemfileIsRequiredForBundler)
 }
+
+// getAllVulnerabilitiesWithDetailsJoined will add the default separator of the details and will check if the hash
+// already exists in list of the vulnerabilities, if is duplicated, so, it will only join both details.
+// nolint:funlen,gocyclo // Breaking this function will make it more confusing
+func (a *Analyzer) getAllVulnerabilitiesWithDetailsJoined() (
+	newAnalysisVulnerabilities []analysis.AnalysisVulnerabilities,
+) {
+	for currentIndex := range a.analysis.AnalysisVulnerabilities {
+		currentAV := a.analysis.AnalysisVulnerabilities[currentIndex]
+		exists := false
+		for newIndex := range newAnalysisVulnerabilities {
+			newAV := newAnalysisVulnerabilities[newIndex]
+			if currentAV.Vulnerability.VulnHash == newAV.Vulnerability.VulnHash &&
+				!strings.Contains(newAV.Vulnerability.Details, currentAV.Vulnerability.Details) {
+				exists = true
+				newAnalysisVulnerabilities[newIndex].Vulnerability.Details = fmt.Sprintf("%s\n         %s%s",
```

Comment:
Shouldn't be `newAV.Vulnerability.Details = fmt.Sprintf(...)` ?

Reply:
I tried to change it but as it is inside a loop the reference is not fixed, I believe we will have to keep it that way

Reply:
I thought that was a slice of pointers.